gluten-core/src/main/java/org/apache/gluten/jni/JniWorkspace.java

```diff
@@ -99,12 +85,17 @@ public static void enableDebug(String debugDir) {
     }
   }
 
-  public static JniWorkspace getDefault() {
+  public static void initializeDefault(String rootDir) {
     synchronized (DEFAULT_INSTANCE_INIT_LOCK) {
       if (DEFAULT_INSTANCE == null) {
-        DEFAULT_INSTANCE = createDefault();
+        DEFAULT_INSTANCE = createOrGet(rootDir);
       }
-      Preconditions.checkNotNull(DEFAULT_INSTANCE);
+    }
+  }
+
+  public static JniWorkspace getDefault() {
+    synchronized (DEFAULT_INSTANCE_INIT_LOCK) {
```

Comment:
Remove `synchronized`.

Reply:
ditto

Reply:
@shuai-xu, seems no need to add "synchronized" in getting the instance.